### PR TITLE
JBPM-9254 Stunner - BPMN Editor freezes in business central

### PR DESCRIPTION
**Pull request by hasys**
**JIRA**: [JBPM-9254](https://issues.redhat.com/browse/JBPM-9254)

**Business Central**: [here](https://drive.google.com/file/d/1jH__H8IhyHAhXlLWnqtqmL55wJ-omdU2/view?usp=sharing)

**VS Code**: [here](https://drive.google.com/file/d/1l292hIuhDJQwMdv-DIzm9d0_nZUBbHjD/view?usp=sharing)

<details>
<summary>
Hi @romartin, there is a fix for the jira, I tested it with business central and I can't reproduce it anymore. See details under the cut.
</summary>

<br>

<b>A side effect</b>: now if user will manually declare twice the same variable like `userVar[String]->processVar` all duplicates will be removed. But it won't remove variables if only one or two of tree parts (variable name, variable type, variable association) are the same, like `userVar[String]->processVar1` and `userVar[String]->processVar2` will be preserved.

Fix can be more elegant than adding addition `Set` to keep names of processed variables but it will need a bit more deep rework of marshallers due to missing overriding of `equals` method in EMF library. Since we are planning to replace marshallers by new one, I think this rework is unnecessary and this tmp `Set` should be enough. WDYT? If you agree with this solution I will duplicate solution for Kogito version of marshallers and attach Binaries for further testing. Thank you!

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Review by romartin — APPROVED** [review]
Thanks @hasys , this solution looks good enough to move forward, so let's go ahead with this PR!! :+1:

**Comment by hasys** [comment]
jenkins do fdb

**Comment by romartin** [comment]
ok @hasys thanks for the updates. Lemme just clarify plz - so I understand this issue finally affects both backend (BC) and client (kogito) side marshallers, so the fix is done on both sides on the code too. Is that correct?

**Comment by hasys** [comment]
jenkins do fdb

**Comment by hasys** [comment]
Hi @romartin, there are two issues:
1. Generated huge file
2. in Business Central validation/indexation mechanism causes freeze of Business central

The first one causes the second one and present in both Kogito and Business Central. But the second issue appears in Business Central only.

This PR fixing 1-st issue for both Kogito and Business Central which is the root cause of the second issue (from the issue title).

So in short: freeze not happens in Kogito, but the root cause of the freeze was present in both in Kogito and in Business Central and now fixed also in both places :)

**Comment by hasys** [comment]
jenkins do fdb

**Comment by hasys** [comment]
Sonar cloud failing due to duplications in Kogito and Business Central marshallers.

**Review by domhanak — APPROVED** [review]
Looks good to me. Tested in business central and VSCode editor.

Files linked to JIRA can be now fixed in BC or VSCode.
Duplicate data outputs are removed upon saving.
